In [1]:
import pandas as pd
import matplotlib

First we'll import the data and and just look at some of its attributes

In [2]:
df = pd.read_csv("../data/county_broadband_adoption.csv")
print(df.shape)
df.sample(10)

(59527, 6)


statenam               county  year              id  broadband  \
113          Alabama       Bullock County  2018  0500000US01011   0.589000   
56390     Washington      Okanogan County  2017  0500000US53047   0.431095   
12600       Illinois         Piatt County  2003  0500000US17147        NaN   
56718  West Virginia         Boone County  2003  0500000US54005        NaN   
48282          Texas      Burleson County  2003  0500000US48051        NaN   
4475      California        Tehama County  2010  0500000US06103        NaN   
6536         Florida  Hillsborough County  2000  0500000US12057   0.079695   
3292        Arkansas     Sebastian County  2005  0500000US05131        NaN   
35107       New York    Livingston County  2014  0500000US36051   0.789125   
24330       Michigan     Menominee County  2010  0500000US26109        NaN   

       cfips  
113     1011  
56390  53047  
12600  17147  
56718  54005  
48282  48051  
4475    6103  
6536   12057  
3292    5131  
35107  36051  
24330  26109

There are 59527 rows where each row represents a county of a state in a certain year and the broadband adoption percentage of that county in that year. A cleaner dataframe would be as such.

In [3]:
columns = {
    "statenam": "state",
    "broadband": "adoption_pct"
}
df2 = df[["statenam", "county", "year", "broadband"]].dropna().rename(columns=columns)
df2.head(20)

state          county  year  adoption_pct
17  Alabama  Autauga County  2017      0.618182
18  Alabama  Autauga County  2018      0.789000
26  Alabama  Baldwin County  2007      0.588521
27  Alabama  Baldwin County  2008      0.591518
28  Alabama  Baldwin County  2009      0.594515
29  Alabama  Baldwin County  2010      0.730085
30  Alabama  Baldwin County  2011      0.743012
31  Alabama  Baldwin County  2012      0.819295
32  Alabama  Baldwin County  2013      0.718728
33  Alabama  Baldwin County  2014      0.747387
34  Alabama  Baldwin County  2015      0.734921
35  Alabama  Baldwin County  2016      0.771539
36  Alabama  Baldwin County  2017      0.820682
37  Alabama  Baldwin County  2018      0.781000
55  Alabama  Barbour County  2017      0.478992
56  Alabama  Barbour County  2018      0.604000
74  Alabama     Bibb County  2017      0.606742
75  Alabama     Bibb County  2018      0.661000
93  Alabama   Blount County  2017      0.586694
94  Alabama   Blount County  2018      0.685000

In [4]:
meanstatepct2018 = df2[df2["year"] == 2018].groupby(["state"]).mean()
meanstatepct2018.plot.bar(y="adoption_pct", rot=0)

In [5]:
meanstatepct2018sorted = meanstatepct2018.sort_values(by="adoption_pct")
meanstatepct2018sorted.head(5)

year  adoption_pct
state                          
Mississippi  2018      0.611183
New Mexico   2018      0.638667
Arkansas     2018      0.641640
Alabama      2018      0.657627
Louisiana    2018      0.658609

In [6]:
meanstatepct2018sorted.tail(5)

year  adoption_pct
state                            
New Hampshire  2018      0.835800
New Jersey     2018      0.837429
Massachusetts  2018      0.838429
Rhode Island   2018      0.838600
Connecticut    2018      0.841250

There's an overall increasing trend of internet adoption over all counties. We can try to run a regression analysis if we can find similarly grained data regarding education statistics by county. However, we can try to utilize trends by counties and states to also tell stories if similarly grained data cannot be found.

# FB

* Add some more of the basic descriptives,
    * how many years
    * how many counties 
   
* And make some observations on these.


* But from the connectivity standpoint this looks like great data as one of the core elements for your analysis. Does educational achievement data exist of the same time period and at the same county level?

In [7]:
df['year'].unique()

array([2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010,
       2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018])

In [8]:
df['year'].nunique()

19

* So max is 19 years of data for each county

In [9]:
df['county'].nunique()

1867

In [12]:
1867*19

35473

In [14]:
df.shape[0]

59527

* Why more rows in df that counties multiplied by the number of years?

In [16]:
df.groupby('county').size().sort_values(ascending=False).head(20)

county
Washington County    570
Jefferson County     475
Franklin County      456
Lincoln County       437
Jackson County       437
Madison County       361
Clay County          342
Montgomery County    342
Marion County        323
Monroe County        323
Union County         323
Wayne County         304
Warren County        266
Grant County         266
Greene County        266
Carroll County       247
Douglas County       228
Marshall County      228
Polk County          228
Johnson County       228
dtype: int64

* So it looks like `county` is not the smallest unit in this data?

* So need to aggregate up to `county` level or look at the `fips` or `id` columns you dropped.
